### PR TITLE
Update ESLint and Rubocop configs

### PR DESCRIPTION
**Pull request by mkostick**
Both of these were slightly incompatible with the current versions of
these tools.  I used the warning outputs from Code Climate to identify
these, and many of the warnings suggested what to replace the deprecated
settings with.

I also enabled a few Rubocop style checkers for whitespace checks that I
think will not be controversial

### CODE REVIEW

File: config/.eslintrc.yml [hunk 1]
```diff
@@ -181,7 +185,10 @@ rules:
   semi-spacing: 0
   semi: 0
   sort-vars: 0
-  space-after-keywords: 0
+  keyword-spacing:
+  - 0
+  - after: false
+    before: false
```
Comment (ancharon) [inline config/.eslintrc.yml]: Does this need a `-` before it?

File: config/.rubocop.yml [hunk 2]
```diff
@@ -1090,22 +1081,27 @@ Style/SymbolProc:
...
 Style/TrailingBlankLines:
   Description: 'Checks trailing blank lines and final newline.'
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#newline-eof'
   Enabled: false
 
-Style/TrailingComma:
-  Description: 'Checks for trailing comma in parameter lists and literals.'
+Style/TrailingCommaInArguments:
+  Description: 'Checks for trailing comma in parameter lists.'
+  StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#no-trailing-params-commas'
```
Comment (ancharon) [inline config/.rubocop.yml]: This anchor doesn't go anywhere -- outdated link?

Comment (ancharon) [inline config/.rubocop.yml]: Looks like maybe https://github.com/bbatsov/ruby-style-guide#no-trailing-params-comma is the right one

File: config/.eslintrc.yml (hunk 1 above)
Comment (mkostick) [inline config/.eslintrc.yml]: No, the after/before pair are keys of the same hash.

**Pull request merged**